File: common/src/main/java/generations/gg/generations/core/generationscore/world/level/block/decorations/HouseLampBlock.java
package generations.gg.generations.core.generationscore.world.level.block.decorations;

import generations.gg.generations.core.generationscore.world.level.block.entities.GenerationsBlockEntities;
import generations.gg.generations.core.generationscore.world.level.block.entities.PokeModBlockEntityModels;
import generations.gg.generations.core.generationscore.world.level.block.entities.generic.GenericModelProvidingBlockEntity;
import generations.gg.generations.core.generationscore.world.level.block.generic.GenericRotatableModelBlock;
import net.minecraft.world.level.block.state.BlockBehaviour;

public class HouseLampBlock extends GenericRotatableModelBlock<GenericModelProvidingBlockEntity> {
    public HouseLampBlock(BlockBehaviour.Properties props) {
        super(props, GenerationsBlockEntities.GENERIC_MODEL_PROVIDING, PokeModBlockEntityModels.HOUSE_LAMP);
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/level/block/decorations/HouseLampBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block.decorations

import com.mojang.serialization.MapCodec
import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntities
import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntityModels
import generations.gg.generations.core.generationscore.common.world.level.block.generic.GenericRotatableModelBlock

class HouseLampBlock(properties: Properties) :
    GenericRotatableModelBlock(
        properties = properties,
        model = GenerationsBlockEntityModels.HOUSE_LAMP
    ) {
    override val blockEntityType
        get() = GenerationsBlockEntities.GENERIC_MODEL_PROVIDING

    override fun codec(): MapCodec<HouseLampBlock> = CODEC

    companion object {
        val CODEC: MapCodec<HouseLampBlock> = simpleCodec(::HouseLampBlock)
    }
}